src/lib/runner.rs

```diff
@@ -129,26 +128,27 @@ pub fn generate(missile: &Missile, launch_parameters: &LaunchParameter, timestep
 		let burn_1 = burn_0.end..missile.timefire1;
 
 		let burn_time = f64::from(i) * timestep;
-		let compute_mass = |mass_end, timefire| {
-			missile.mass + (missile.mass - mass_end) * (burn_time/ timefire)
+		let compute_mass = |delta_mass, mass_end, timefire| {
+			delta_mass - ((delta_mass - mass_end) / timefire * burn_time)
 		};
 		match () {
 			_ if burn_0.contains(&burn_time) => {
-				mass = compute_mass(missile.mass_end, missile.timefire0);
+				mass = compute_mass(missile.mass, missile.mass_end, missile.timefire0);
 				force = missile.force0;
 				engine_stage = "0";
 			}
 			_ if burn_1.contains(&burn_time) => {
-				mass = compute_mass(missile.mass_end1, missile.timefire1);
+				mass = compute_mass(missile.mass_end, missile.mass_end1, missile.timefire1);
 				force = missile.force1;
 				engine_stage = "1";
 			}
 			_ => {
-				mass = missile.mass_end;
+				mass = missile.mass_end1;
```

Comment:
will the fix work for non two-stage missiles? or will they receive 0kg mass?

Reply:
This would overwrite the mass to always be mass_end1 (which is 0 when there is no 2nd stage)

The match statement checks if the current time is within burn phase 0 or 1, burn phase 1 will be of length 0 if there is no 2nd stage engine. This case should be covered.

Reply:
`thread 'main' panicked at 'byte index 5 is out of bounds of `0`', src/lib\runner.rs:194:22`
![image](https://user-images.githubusercontent.com/17598243/152402688-74f20d47-d017-4e3b-aa1c-1cf4b153386e.png)
for **aim7e2**
looks like mass is set to 0 however thats not the problem of the crash

Reply:
setting mass for missile like
`    "mass": 1.0,
    "mass_end": 1.0,
    "mass_end1": 1.0,
`
will reproduce the bug